src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -88,30 +103,23 @@
      * @param gooeyComponent
      */
     @ReceiveEvent
-    public void onBreedingGooey(BreedGooeyEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent, HungerComponent hungerComponent) {
-        CharacterHeldItemComponent characterHeldItemComponent = event.getInstigator().getComponent(CharacterHeldItemComponent.class);
-
-        if (characterHeldItemComponent != null && characterHeldItemComponent.selectedItem.getComponent(DisplayNameComponent.class) != null) {
-            EntityRef item = characterHeldItemComponent.selectedItem;
-            String itemName = item.getComponent(DisplayNameComponent.class).name;
-
-            if (!itemName.isEmpty() && hungerComponent.food.contains(itemName)) {
-                logger.info("Selected for breeding ...");
-                FollowComponent followComponent = gooeyEntity.getComponent(FollowComponent.class);
-                BehaviorComponent behaviorComponent = gooeyEntity.getComponent(BehaviorComponent.class);
-                MatingComponent matingComponent = new MatingComponent();
-
-                followComponent.entityToFollow = event.getInstigator();
-                behaviorComponent.tree = assetManager.getAsset("GooKeeper:breedingBehavior", BehaviorTree.class).get();
-                CharacterMovementComponent characterMovementComponent = gooeyEntity.getComponent(CharacterMovementComponent.class);
-                characterMovementComponent.jumpSpeed = 12f;
-
-                gooeyEntity.saveComponent(followComponent);
-                gooeyEntity.saveComponent(behaviorComponent);
-                gooeyEntity.addOrSaveComponent(matingComponent);
-                gooeyEntity.saveComponent(characterMovementComponent);
+    public void onBreedingGooey(BreedGooeyEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent) {
+        logger.info("Selected for breeding...");
+        MatingComponent matingComponent = new MatingComponent();
+
+        matingComponent.selectedForMating = !matingComponent.selectedForMating;
+
+        for (EntityRef breedingBlock : entityManager.getEntitiesWith(BreedingBlockComponent.class)) {
```

Comment:
This for loop should not get executed if selectedForMating is false? From what I understand, as it is right now, if the player deselcts a gooey for mating, line 110 will toggle `selectedForMating` to false (which is right), but this `for` loop will still get executed (which you do not want) and thus select a partner for mating

Reply:
Maybe you want to enclose this whole loop inside `if (matingComponent.selectedForMating) {...}`